stan/math/rev/fun/read_corr_L.hpp

```diff
@@ -0,0 +1,179 @@
+#ifndef STAN_MATH_REV_FUN_READ_CORR_L_HPP
+#define STAN_MATH_REV_FUN_READ_CORR_L_HPP
+
+#include <stan/math/prim/fun/read_corr_L.hpp>
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/core/operator_multiplication.hpp>
+#include <cmath>
+#include <complex>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the Cholesky factor of the correlation matrix of the
+ * specified dimensionality corresponding to the specified
+ * canonical partial correlations.
+ *
+ * It is generally better to work with the Cholesky factor rather
+ * than the correlation matrix itself when the determinant,
+ * inverse, etc. of the correlation matrix is needed for some
+ * statistical calculation.
+ *
+ * <p>See <code>read_corr_matrix(Array, size_t, T)</code>
+ * for more information.
+ *
+ * @tparam T type of input (must be a `var_value<T>` where `T`
+ *  inherits from EigenBase)
+ * @param CPCs The (K choose 2) canonical partial correlations in
+ * (-1, 1).
+ * @param K Dimensionality of correlation matrix.
+ * @return Cholesky factor of correlation matrix for specified
+ * canonical partial correlations.
+ */
+template <typename T, require_var_vector_t<T>* = nullptr>
+auto read_corr_L(const T& CPCs, size_t K) {  // on (-1, 1)
+  using ret_type = var_value<Eigen::MatrixXd>;
+
+  if (K == 0) {
+    return ret_type(Eigen::MatrixXd());
+  }
+  if (K == 1) {
+    return ret_type(Eigen::MatrixXd::Identity(1, 1));
+  }
+
+  using std::sqrt;
+  Eigen::ArrayXd acc(K - 1);
+  acc.setOnes();
+  // Cholesky factor of correlation matrix
+  Eigen::MatrixXd L = Eigen::MatrixXd::Zero(K, K);
```

Comment:
Similarly this can be an `arena_t<Eigen::MatrixXd>` then when you make `L_res` (should that be `L_ret` for return?) it will not make a copy